### ISSUE TITLE
mssql won't install behind corporate proxy on MacOS

### ISSUE THREAD

**Issue by christianbalzer** [Microsoft/vscode-mssql#779]
I installed Visual Studio Code and the mssql extension today - or rather, I tried.

- MSSQL Extension Version: cf. error below
- VSCode Version: 1.10.2
- OS Version: MacOS 10.12.3

Steps to Reproduce:

1. Install plugin, open SQL file. Output pane pops up with output below:

```
Initializing SQL tools service for the mssql extension.
Note: mssql commands will be available after installing the service.

Platform: darwin, x86_64 (OSX)

Installing SQL tools service to /Users/balzec01/.vscode/extensions/ms-mssql.mssql-0.3.0/sqltoolsservice/0.3.0.5/OSX.
Downloading https://download.microsoft.com/download/A/A/2/AA22B2D6-94D8-4C82-A388-D0246F567BBF/microsoft.sqltools.servicelayer-osx-x64-netcoreapp1.0.tar.gz
[ERROR] Error: Request error: EPROTO
```

Help, please? :)

**Comment by kburtram** [comment]
@llali can you confirm if this is the same as the issue you recently fixed?

**Comment by llali** [comment]
Yes probably related to the issue I recently fixed. @christianbalzer please try the latest daily release and if you still have the same problem let me know. you just need to make some changes in the extension before you run it. it's all explained here: https://github.com/Microsoft/vscode-mssql/issues/568
I'll create a temporary release that points to the url in Microsoft download center soon but for now you can follow these steps:

You need to download this url: https://github.com/Microsoft/vscode-mssql/releases/download/v0.3.0-alpha.13/mssql-0.3.0.vsix

The build is downloading the service from github release so you need to change the configuration to get the service from Microsoft download center. After you install the extension please change the config file here:

Mac: ~/.vscode/extensions/ms-mssql.mssql-0.3.0/out/src/config.json
Windows: %USERPROFILE%/.vscode/extensions/ms-mssql.mssql-0.3.0/out/src/config.json

in config.json you need to change the value of "downloadUrl" to:

https://download.microsoft.com/download/A/A/2/AA22B2D6-94D8-4C82-A388-D0246F567BBF/microsoft.sqltools.servicelayer-{#fileName#}

**Comment by christianbalzer** [comment]
> Initializing SQL tools service for the mssql extension.
> Note: mssql commands will be available after installing the service.
> 
> Platform: darwin, x86_64 (OSX)
> 
> Installing SQL tools service to /Users/.../.vscode/extensions/ms-mssql.mssql-0.3.0/sqltoolsservice/0.3.0-alpha.14/OSX.
> Downloading https://download.microsoft.com/download/A/A/2/AA22B2D6-94D8-4C82-A388-D0246F567BBF/microsoft.sqltools.servicelayer-osx-x64-netcoreapp1.0.tar.gz
> (22761 KB) .................... Done!
> Installing ...
> Done! 155 files unpacked.

Installs. Thanks. :)

**Comment by christianbalzer** [comment]
Having said that, the SQL server is this side of the proxy, but now it looks like all connections go through the proxy - so while I can install the plugin, I can't use it... :-/

**Comment by kevcunnane** [comment]
@christianbalzer - For the server you're connecting to, can you use the IP address as the server name in order to connect? 
Also, you can verify that connections work as expected by testing against a local SQL Server instance (on docker). See [here](https://docs.microsoft.com/en-us/sql/linux/sql-server-linux-setup-docker) for instructions on setting this up.

**Comment by batzen** [comment]
I am also suffering from this issue (extension is not being able to download the service tools).

What's your suggested way to solve this without involving manual installation and modification steps?
The C# extension has the same issue...

**Comment by kevcunnane** [comment]
@batzen we recommend downloading a fully-packaged version of the VSIX and using `Extensions: Install from VSIX...` command to install it. The 1.0 release is [here](https://github.com/Microsoft/vscode-mssql/releases/tag/v1.0.0) and our upcoming 1.1 release will also be on Github. You can download for the relevant platform and use this - e.g. use [this for windows x64](https://github.com/Microsoft/vscode-mssql/releases/download/v1.0.0/mssql-1.0.0-win7-x64.vsix).

**Comment by batzen** [comment]
@kevcunnane I don't know much about extension development for vs code, but why doesn't the extension do this for me? If the extension can't do this for me it would be nice to get a message stating what to do instead of just failing to download. The same recommendation you gave here would be a step forward.
Currently people only get a generic error and don't know what to do next.

**Comment by llali** [comment]
@batzen the extension already handles downloading with proxy as long as the proxy setting is setup correctly in vs code. there was a bug in mssql extension which is fixed now and if you use the latest version, you should be able to download the service. If c# extension and mssql both have problems, it might be your proxy setting is wrong
